docs/user-guide/adaptive-batching.md

```diff
@@ -0,0 +1,73 @@
+# Adaptive Batching
+
+MLServer includes support to batch requests together transparently on-the-fly.
+We refer to this as "adaptive batching", although it can also be known as
+"predictive batching".
+
+![](../assets/adaptive-batching.svg)
+
+## Benefits
+
+There are usually two main reasons to adopt adaptive batching:
+
+- **Maximise resource usage**.
+  Usually, inference operations are “vectorised” (i.e. are designed to operate
+  across batches).
+  For example, a GPU is designed to operate on multiple data points at the same
+  time.
+  Therefore, to make sure that it’s used at maximum capacity, we need to run
+  inference across batches.
+
+- **Minimise any inference overhead**.
+  Usually, all models will have to _“pay”_ a constant overhead when running any
+  type of inference.
+  This can be something like IO to communicate with the GPU or some kind of
+  processing in the incoming data.
+  Up to a certain size, this overhead tends to not scale linearly with the
+  number of data points.
+  Therefore, it’s in our interest to send as large batches as we can without
+  deteriorating performance.
+
+However, these benefits will usually scale only up to a certain point, which is
+usually determined by either the infrastructure, the machine learning
+framework used to train your model, or a combination of both.
+Therefore, to maximise the performance improvements brought in by adaptive
+batching it will be important to [configure it with the appropiate values for
+your model](#usage).
+Since these values are usually found through experimentation, **MLServer won't
+enable by default adaptive batching on newly loaded models**.
+
+## Usage
+
+MLServer lets you configure adaptive batching independently for each model
+through two main parameters:
+
+- **Maximum batch size**, that is how many requests you want to group together.
```

Comment:
Interesting that Triton has max_batch_size as a core settings outside adaptive batching. I suppose this is for memory reasons to not allow too big a request irrespective of if its in a single request or in an adaptive batch

Reply:
So can users set `max_batch_size` outside adaptive batching?

Reply:
No, not really. This would only apply for adaptive batching. Outside of it, the model accepts whatever it gets. 

In the case of Triton, do you mean that it will split a request into smaller chunks if it's too large? That's interesting.

Reply:
I think in Triton its a head limit on max batch a server can consume. I assume it returns an error if you send a batch too large.